src/githelp/overlays.py

```diff
@@ -0,0 +1,177 @@
+import os
+import yaml
+
+#build absolute path to githelp/data/overlays
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+DATA_DIR = os.path.join(THIS_DIR, "data", "overlays")
+#master dictionary file: to /src/githelp/data/overlays/tips.yml
+MASTER_FILE = os.path.join(DATA_DIR, "tips.yml")
+
+
+def read_yaml(path: str):
+    '''
+    Read a YAML file into a Python object.
+
+    Parameters
+    ----------
+    path : str
+        Path to the YAML file to read.
+
+    Returns
+    -------
+    dict
+        Parsed YAML contents. Returns an empty dict if the file is empty
+        or if any error occurs while reading/parsing.
+    '''
+    try:
+        #r opens the file for reading in text mode, and encoding="utf-8" tells Python to decode the file's bytes 
+        #as UTF-8 text into normal Python strings
+        with open(path, "r", encoding="utf-8") as f:
+            #convert YAML content to Python object or empty dict if file is empty
+            data = yaml.safe_load(f) or {}
+        #return the data of whatever was loaded
+        return data
+    except Exception:
+        #if error occurs return empty dict
+        return {}
+    
+def helper():
+
+    data = read_yaml(MASTER_FILE)
+    return data
+
+def load_overlay(command: str):
+    '''
+    Get the tips for one git subcommand.
+
+    This first looks in the master file tips.yml under the given
+    command name (ex "pull"). If found, it returns the corresponding
+    dictionary of tips.
+
+    Parameters
+    ----------
+    command : str
+        Name of the git subcommand whose overlay should be loaded.
+
+    Returns
+    -------
+    dict or None
+        The overlay dictionary for the given command, with a ``"command"`` key
+        ensured, or ``None`` if no overlay is defined.
+    '''
+    #set data to the master dictionary file
+    data = helper()
+    #check to see if the master is a dict and not empty
```

Comment:
again, remove references to master and why not jsut use the read function directly?